docs/_articles/en/v3.x/salesforce-graph-engine/working-with-sfge.md

```diff
@@ -82,36 +82,61 @@ Example:
+2. Decrease the number of parallel threads by setting the `--rule-thread-count` parameter to 2.
+3. Increase timeout by setting the `--rule-thread-timeout` parameter to 300000 ms.
 
-	For example, to allocate 2 G heap space:
+### Knobs to Control Graph Engine Execution
+Two Graph Engine parameters, `--sfgejvmargs` and `--pathexplimit`, act as knobs that turn the max heap size and the complexity of Graph Engine scans up or down. Use these knobs to fine-tune your code’s analysis and rate of `OutOfMemory` occurrences.
 
-	```
-	sfdx scanner:run:dfa --sfgejvmargs "-Xmx2g" <rest of your parameters>
-	```
-	or
-	```
-	export SFGE_JVM_ARGS="-Xmx2g"
-	sfdx scanner:run:dfa <rest of your parameters>
-	```
-	Because the heap space value depends on the complexity of the target codebase, there's no magic number. A very large heap space can degrade Graph Engine’s performance, so increase the heap space allocation in increments of 1 G. Experiment to see what works for your project.
+#### Modify the Allocated Heap Space with `--sfgejvmargs`
+Use the `--sfgejvmargs` parameter to modify your Java Virtual Machine (JVM) default max heap size.
+
+1. Look up your JVM `-Xmx` value, which is your allocated heap size. 
+2. Use the `--sfgejvmargs` parameter to increase your `-Xmx` value on `scanner:run:dfa command`. 
+3. Execute Graph Engine with a larger heap space than the default settings.
+
+For example, to allocate 2 G heap space:
+
+	`sfdx scanner:run:dfa --sfgejvmargs "-Xmx2g" <rest of your parameters>`
+
+To maximize your heap space balance with Graph Engine performance, follow these recommendations.
+
+* Because the heap space value depends on the complexity of the target codebase, there’s no magic number. A very large heap space can degrade Graph Engine’s performance, so increase the heap space allocation in increments of 1 G. Experiment to see what works for your project.
+* Target a smaller set of files for analysis. Provide a subset of Apex files using the `--target` flag on the `scanner:run:dfa`command while keeping the same `--projectdir` value. This approach reduces the number of paths and reduces the likelihood of `OutOfMemory` errors.
+* To avoid large IF/ELSE-IF/ELSE conditional trees, simplify your code, which helps bring down the number of paths created.
+
+#### Set complexity-handling-limit Using`--pathexplimit` Parameter
+Heap space allocated for a `scanner:run:dfa` execution also dictates how much complexity Graph Engine can handle. If you ran our recommended steps earlier, grab the path expansion limit that you looked up earlier.
 
-* Target a smaller set of files for analysis. Provide a subset of Apex files using the `--target` flag on the `scanner:run:dfa` command while keeping the same `--projectdir` value. This approach reduces the number of paths and reduces the likelihood of `OutOfMemory` errors.
+Override your path expansion limit using the `--pathexplimit` parameter. Or remove the limit by passing in this value as “-1”.
```

Comment:
Can we use <backtick>-1<backtick> instead of quotes (“-1”)? There's a good chance users would copy-paste this. The quotes would make the OS treat this as a String, while we want it to treat it as a number.

Reply:
yes